src/Uno.UI/UI/Xaml/Controls/Control/Control.cs

```diff
@@ -434,6 +434,14 @@ private protected override void OnLoaded()
 
 			base.OnLoaded();
 
+			foreach (var visualStateGroup in VisualStateManager.GetVisualStateGroups(GetTemplateRoot()))
+			{
+				foreach (var visualState in visualStateGroup.States)
+				{
+					visualState.EnsureMaterialized();
+				}
+				visualStateGroup.UpdateResourceBindings();
+			}
```

Comment:
Why are we doing this? Looks like we are explicitly disabling a perf optimization here, if it cause a bug we should either fix it, either just remove that optimization.

cc @jeromelaban (apparently you did it).

Reply:
Looking at the code, the `EnsureMaterialized` is invoked when  you access to the `StoryBoard` or the `Setters` of the `VisualState`. You should probably be able to move the `UpdateResourceBindings` directly in that method.

Reply:
Removing the edits in `Control.cs` and adding a call in `EnsureMaterialized` to `UpdateReasourceBindings` somehow works (even though I've tried some variations of that before). To make it more perplexing, after making that adjustment, this doesn't work:

```csharp
private protected override void OnLoaded()
		{
			SetUpdateControlTemplate();

			base.OnLoaded();

			foreach (var visualStateGroup in VisualStateManager.GetVisualStateGroups(GetTemplateRoot()))
			{
				foreach (var visualState in visualStateGroup.States)
				{
					visualState.EnsureMaterialized();
				}
				// visualStateGroup.UpdateResourceBindings();
			}
		}
```

This also doesn't work:
```csharp
private protected override void OnLoaded()
		{
			SetUpdateControlTemplate();

			base.OnLoaded();

			foreach (var visualStateGroup in VisualStateManager.GetVisualStateGroups(GetTemplateRoot()))
			{
				// foreach (var visualState in visualStateGroup.States)
				// {
				// 	visualState.EnsureMaterialized();
				// }
				visualStateGroup.UpdateResourceBindings();
			}
		}
```

Meaning, redundant calls to ` VisualStateGroup.UpdateResourceBindings()` or `VisualState.EnsureMaterialized` somehow break things (even though one would think they're idemponent). BUT, making the redundant calls together works (!!!). I'm not sure what's causing this. I went quite deep into the call stack, but nothing looks alarming.

Reply:
I'm not sure what happened, but I've tested the previous two snippets again, and the second one works, so the behaviour is less crazy than what I previously said. I'll test one more time later and see if everything makes sense.